include/srf/segment/egress_port.hpp

```diff
@@ -61,7 +61,9 @@ class EgressPort final : public Object<node::SinkProperties<T>>,
       m_segment_address(address),
       m_port_name(std::move(name)),
       m_sink(std::make_unique<node::RxNode<T>>())
-    {}
+    {
+        this->m_name = m_port_name;
```

Comment:
`ObjectProperties::name()` can be overridden to return `port_name()`

I'm not sure how `segment::Object` a protected member variable `m_name`, but I think that was an oversight.  I think some cpp python proxy object added it instead of deriving from `segment::Object`.

Reply:
A couple things to note:

1. ObjectProperties::name() is marked final by the Segment::object implementation. Easy enough to remove so that Ingress/Egress can override it, but that doesn't solve all the issues.
2. Runnable sets `this->m_name` in its constructor.
3. A pure virtual setter on ObjectProperties can't be called from within a constructor, so that would mean you'd have to create whatever object, then call obj->set_name("name").

Given the use cases I can see, my current solution, to avoid using protected member variables, is to make `m_name` private, and add a protected `set_name` method to segment::Object. This avoids changing `name()`, eliminates the protected member variable, and allows m_name to be set in the places its used now.